libsolidity/ast/AST.cpp

```diff
@@ -84,13 +84,41 @@ SourceUnitAnnotation& SourceUnit::annotation() const
 	return dynamic_cast<SourceUnitAnnotation&>(*m_annotation);
 }
 
-string Declaration::sourceUnitName() const
+std::set<SourceUnit const*> SourceUnit::referencedSourceUnits(bool _flatten, std::set<SourceUnit const*> _skipList) const
+{
+	std::set<SourceUnit const*> sourceUnits;
+	for (auto const& node: nodes())
```

Comment:
You could use `for (ImportDirective const* importDirective: filteredNodes<ImportDirective>(nodes())) { ... }`